testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -505,6 +508,32 @@ def test_fasta2select_ClustalW(self, tmpdir):
         assert len(
             sel['mobile']) == 23090, "selection string has unexpected length"
 
+    def test_fasta2select_nogap(self, tmpdir):
+        """test align.fasta2select() on aligned FASTA with no gap in residue
+        (Issue #3124)"""
+        reference = mda.Universe(self.ref).select_atoms('all')
+        target = mda.Universe(self.tgt).select_atoms('all')
+        ref_resids = [a.resid for a in
+                      reference.select_atoms('protein').select_atoms(
```

Comment:
it might make it more legible (and also reduce the number of lines to create the atomgroups instead of doing it on the fly in the list comprehension?